src/database.rs

```diff
@@ -716,6 +716,15 @@ impl Database {
         self.parse_result().statement(addr)
     }
 
+    /// Get the name of the source file for a given statement.
+    #[must_use]
+    pub fn statement_source_name(&self, addr: StatementAddress) -> String {
+        self.parse_result()
+            .source_info(addr.segment_id)
+            .name
```

Comment:
Can we do this without the clone?

Reply:
The `parse_result` segment set is behind an `Arc`. If I don't clone, the borrow checker complains that `name` cannot be moved out of an `Arc`.

I could also return a short-lived reference like so:
```rust
    /// Get the name of the source file for a given statement.
    #[must_use]
    pub fn statement_source_name(&self, addr: StatementAddress) -> &'_ str {
        self.parse_result()
            .source_info(addr.segment_id)
            .name
            .as_ref()
    }
```